openml/config.py

```diff
@@ -19,7 +19,7 @@
     'apikey': None,
     'server': "https://www.openml.org/api/v1/xml",
     'verbosity': 0,
-    'cachedir': os.path.expanduser('~/.openml/cache'),
+    'cachedir': os.path.expanduser(os.path.join('~', '.openml', 'cache')),
```

Comment:
In light of your change of `/` to `os.sep`, you should replace the `~` by `os.path.expanduser`.

Reply:
@mfeurer according to the docs, "On Unix and Windows, return the argument with an initial component of ~ or ~user replaced by that user’s home directory.". So the ~ is necessary.

Reply:
You're right, my mistake.